### PR TITLE
[CL]: Increase min tick past canonical tick range

### PR DESCRIPTION
**Pull request by AlpinYukseloglu**
Closes: #5550

## What is the purpose of the change

This PR moves the min tick & min spot price for CL up to past the point where canonical ticks start to matter.

Due to a majority of test vectors relating to min tick or full range being hard coded and/or manually computed, this change broke many more tests than we originally expected. Most of them have been amended and modularized to not break for future changes, and the remaining ones were recomputed with the math documented in comments

## Testing and Verifying

All relevant tests were amended to refer to the new min tick/min price. Where applicable, `types.MinTick` was used in place of hard coded values to make future changes break fewer tests.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/gamm/keeper/migrate_test.go [hunk 1]
```diff
@@ -17,8 +17,27 @@ func (s *KeeperTestSuite) TestMigrate() {
...
+	// # Using scripts in x/concentrated-liquidity/python/swap_test.py, we compute the following:
+	// expectedAsset0 = floor((liquidity * (maxSqrtPrice - curSqrtPrice)) / (maxSqrtPrice * curSqrtPrice)) = 99999999999.000000000000000000
+	// expectedAsset1 = floor(liquidity * (curSqrtPrice - minSqrtPrice)) = 99999900000.000000000000000000
+	//
+	// We add 1 to account for ExitPool rounding exit amount up. This is not an issue since the balance is deducted from the user regardless.
+	// These leaves us with full transfer of asset 0 and a (correct) transfer of asset 1 amounting to full GAMM balance minus 100000.
+	// We expect this tolerance to be sufficient as long as our test cases are on the same order of magnitude.
 	defaultErrorTolerance := osmomath.ErrTolerance{
-		AdditiveTolerance: sdk.NewDec(100),
+		AdditiveTolerance: sdk.NewDec(100000),
```
Comment (AlpinYukseloglu) [inline x/gamm/keeper/migrate_test.go:40]: Note to reviewer: sorry for the large comment here, but wanted to make sure our tolerances aren't arbitrary.

A good sanity check for this change is that increasing min price by `10^6` increases min sqrt price by `10^3`, so it follows that liquidity calculations (most of which are done in sqrt price domain` should shift by `10^3`

File: x/protorev/keeper/epoch_hook_test.go [hunk 2]
```diff
@@ -162,7 +162,7 @@ func (s *KeeperTestSuite) TestUpdateHighestLiquidityPools() {
 			},
 			expectedBaseDenomPools: map[string]map[string]keeper.LiquidityPoolStruct{
 				"epochTwo": {
-					"uosmo": {Liquidity: sdk.Int(sdk.NewUintFromString("999999999000000001000000000000000000")), PoolId: 49},
+					"uosmo": {Liquidity: sdk.Int(sdk.NewUintFromString("999999000000000001000000000000000000")), PoolId: 49},
```
Comment (AlpinYukseloglu) [inline x/protorev/keeper/epoch_hook_test.go:165]: Note to reviewer: this number seemed right based on the migration calculations (above) and superfluid calculations (below), but admittedly I did slightly eyeball this as the underlying test components are quite obscure

(there's 0 documentation on where this number comes from, although I can infer it's the asset1 amount for a set of full range positions)

File: x/concentrated-liquidity/tick_test.go [hunk 3]
```diff
@@ -1405,89 +1405,3 @@ func (s *KeeperTestSuite) TestValidateTickRangeIsValid() {
 		})
 	}
 }
-
-func (s *KeeperTestSuite) TestRoundTickToCanonicalPriceTick() {
```
Comment (p0mvn) [inline x/concentrated-liquidity/tick_test.go:1409]: Why is this removed? Is there a replacement for these test vectors?

**Review by p0mvn — APPROVED** [review]
LGTM

File: x/concentrated-liquidity/tick_test.go (hunk 3 above)
Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/tick_test.go:1409]: The notion of canonical price ticks no longer exist, so these vectors no longer make sense. All the test vectors added relating to the behavior in ticks surrounding the new MinTick ensure the appropriate properties hold.

File: x/concentrated-liquidity/lp_test.go [hunk 4]
```diff
@@ -835,8 +828,11 @@ func (s *KeeperTestSuite) TestAddToPosition() {
 				// 1998976eth (amount withdrawn with rounded down amounts) + 998977(token amount in)
 				amount0Expected: sdk.NewInt(2997953),
 				// tokens Provided for token1 is 9999999999 (amount withdrawn) + 5000000000 = 14999999999usdc.
-				// we calcualte calc amount1 by using: https://www.wolframalpha.com/input?i=70.728769315114743567+*+212041526.154556192320661969
-				amount1Expected: sdk.NewInt(14997436189),
+				// We calcualte calc amount1 by using the following equation:
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/lp_test.go:831]:
```suggestion
				// We calculate calc amount1 by using the following equation:
```

File: x/concentrated-liquidity/lp_test.go [hunk 5]
```diff
@@ -835,8 +828,11 @@ func (s *KeeperTestSuite) TestAddToPosition() {
...
+				// liq * (sqrtPriceB - sqrtPriceA), where liq is equal to the original joined liq + added liq, sqrtPriceB is current sqrt price, and sqrtPriceA is min sqrt price.
+				// Note that these numbers were calculated using `GetLiquidityFromAmounts` and `TickToSqrtPrice` and thus assume correctness of those functions.
+				// https://www.wolframalpha.com/input?i=212041526.154556192317664016+*+%2870.728769315114743566+-+0.000001000000000000%29
+				amount1Expected: sdk.NewInt(14997435977),
```
[the remark this answers is not in the document]
Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/lp_test.go:835]: Ah yes sorry – the expected output is bankers rounded instead of truncated due to `GetLiquidityFromAmounts` being bankers rounded. This does not pose any security risks as the balance is still deducted from the caller.

File: x/concentrated-liquidity/math/tick_test.go [hunk 6]
```diff
@@ -505,49 +507,41 @@ func (suite *ConcentratedMathTestSuite) TestTickToSqrtPricePriceToTick_InverseRe
...
-		"smallest + min price * increment 10^17": {
+		"smallest + min price * increment 10^11": {
 			price:        sdk.MustNewDecFromStr("0.100000000000000000"),
 			tickExpected: -9000000,
 		},
-		"smallest + min price * increment 10^18": {
+		"smallest + min price * increment 10^12": {
 			price:        sdk.MustNewDecFromStr("1.000000000000000000"),
 			tickExpected: 0,
 		},
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/math/tick_test.go:547]: Does this title reflect this test case? I think the previous one might have been incorrect as well right?

**Review by czarcas7ic — APPROVED** [review]

**Review by ValarDragon — APPROVED** [review]
This LGTM

**Comment by AlpinYukseloglu** [comment]
Addressed all comments, will merge once CI passes
